### PR TITLE
Fix infinite recursion in remove/replace for #load'ed files...

### LINKED ISSUE #6698: Cyclic #load hangs VS
1. Create a csx file containing only a `#load` of itself (I used the absolute path)
 2. In the C# Interactive window, `#load` the csx file

### PR DESCRIPTION
**Pull request by KevinH-MS**
Fixes #6698

### CODE REVIEW

**Comment by KevinH-MS** [comment]
FYI @amcasey 
/cc @dotnet/roslyn-interactive @dotnet/roslyn-compiler

File: src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs [hunk 1]
```diff
@@ -405,16 +405,16 @@ public SyntaxAndDeclarationManager RemoveSyntaxTrees(HashSet<SyntaxTree> trees)
                 if (directive.ResolvedPath != null)
                 {
                     SyntaxTree loadedTree;
-                    if (TryGetLoadedSyntaxTree(loadedSyntaxTreeMap, directive, out loadedTree))
+                    if (TryGetLoadedSyntaxTree(loadedSyntaxTreeMap, directive, out loadedTree) && !removeSet.Contains(loadedTree))
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs]: `&& removeSet.Add(loadedTree)` and delete add below?

File: src/Compilers/CSharp/Test/Symbol/Compilation/LoadDirectiveTests.cs [hunk 2]
```diff
@@ -140,5 +140,49 @@ public void ErrorInInactiveRegion()
...
+
+            Assert.Equal(2, compilation.SyntaxTrees.Length);
+            compilation.VerifyDiagnostics();
+
+            compilation = compilation.RemoveSyntaxTrees(newTree);
+
+            Assert.Empty(compilation.SyntaxTrees);
+            compilation.VerifyDiagnostics();
+
+            code = "#load \"a.csx\"";
```
Comment (amcasey) [inline src/Compilers/CSharp/Test/Symbol/Compilation/LoadDirectiveTests.cs]: Same as above?

**Comment by amcasey** [comment]
:+1:

File: src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs]: +1

File: src/Compilers/CSharp/Test/Symbol/Compilation/LoadDirectiveTests.cs (hunk 2 above)
Comment (KevinH-MS) [inline src/Compilers/CSharp/Test/Symbol/Compilation/LoadDirectiveTests.cs]: Removed

File: src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs (hunk 1 above)
Comment (KevinH-MS) [inline src/Compilers/CSharp/Portable/Compilation/SyntaxAndDeclarationManager.cs]: Done

**Pull request merged**